### PR TITLE
Move documentation to GitHub Actions, enable preview builds

### PR DESCRIPTION
**Pull request by simonbyrne**
Should make it easier to edit and review docs changes.

- [X] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [X] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by simonbyrne** [comment]
Now available at https://climate-machine.github.io/CLIMA/previews/PR953/

**Comment by charleskawczynski** [comment]
Hm, building on every push might quickly flood our GitHub resources. Maybe this is fine for now, but it might be nice if we migrate to a [comment-trigger](https://github.community/t5/GitHub-Actions/Accessing-commit-message-in-pull-request-event/td-p/40856). Ideally, users could comment `[build-docs]`, or something, to explicitly trigger the doc preview build since not every push will have documentation changes.

**Comment by simonbyrne** [comment]
Hmm, it seems to be building on every pull action (comments, etc), so need to figure this out.

File: .github/workflows/Documenter.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: 1.3
+      - name: Install MPI
+        run: |
+          sudo apt-get update
+          sudo apt-get install mpich libmpich-dev
+      - name: Install dependencies
+        run: julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
```
Comment (fredrikekre) [inline .github/workflows/Documenter.yaml:23]:
```suggestion
        run: julia --project=docs/ -e 'using Pkg; Pkg.instantiate()'
```

Comment (fredrikekre) [reply .github/workflows/Documenter.yaml:23]: Since you commit the manifest you can pre-`dev` the `CLIMA` package as a relative path.

Comment (simonbyrne) [reply .github/workflows/Documenter.yaml]: ah, figured that out.

**Comment by charleskawczynski** [comment]
> Hmm, it seems to be building on every pull action (comments, etc), so need to figure this out.

I didn't notice that because I only tested with one push. Glad you caught that.

**Comment by simonbyrne** [comment]
Actually maybe not. I’m not quite sure what events trigger it

**Comment by fredrikekre** [comment]
Seems to work, thanks for testing.

**Comment by kpamnany** [comment]
These last few are failing because of ECC errors on hpc-22-34, not just the close handle failure. No point in retrying -- that node has to be taken out of service. I suggest opening a ticket with IMSS.

**Comment by charleskawczynski** [comment]
Ok, opened a ticket with IMSS.

**Comment by charleskawczynski** [comment]
Thanks for pointing that out @kpamnany

**Pull request merged**